packages/core/src/lib/implementation/runner.ts

```diff
@@ -27,14 +27,12 @@ export async function executeRunnerConfig(
   });
 
   // read process output from file system and parse it
-  let audits = await readJsonFile<AuditOutputs>(
+  const rawAudits = await readJsonFile<AuditOutputs>(
```

Comment:
Hmm, we're missing an important step here - validating the contents of the JSON file (optimistically casting as `AuditOutputs` assumes something we don't know to be true). There's needs to be `auditOutputsSchema.parse(...)` call here somewhere (and in `executeRunnerFunction` as well). Note that if `outputTransform` is passed, the result of that function should be asserted, because the JSON could contain anything in this case (`rawAudits` is a misleading name, actually, I'd go with `outputs`).

Reply:
The parsing happens in `execute-plugin` function so I changed the type of `audits` to `unknown` within the `runner`.

Reply:
I see, so it was just that the types were lying :sweat_smile: